python/sqlalchemy/security/sqlalchemy-execute-raw-query.py

```diff
@@ -198,6 +198,7 @@
 # ok: sqlalchemy-execute-raw-query
 engine = create_engine('postgresql://user@localhost/database')
 query = select(literal_column("users.fullname", String) + ', ' + literal_column("addresses.email_address").label("title")).where(and_(literal_column("users.id") == literal_column("addresses.user_id"), text("users.name BETWEEN 'm' AND 'z'"), text("(addresses.email_address LIKE :x OR addresses.email_address LIKE :y)"))).select_from(table('users')).select_from(table('addresses'))
+# deepruleid: sqlalchemy-execute-raw-query
 conn.execute(query, {"x":"%@aol.com", "y":"%@msn.com"}).fetchall()
```

Comment:
I am not sure if this is true; I'll have to inspect this later this week

Reply:
Why does this only match with deep?

Reply:
I believe `query` gets sym prop'd (i.e `query = select(...) + ...`) so then you'd get `conn.execute(query, ...)` which might match to this part of the rule?

https://github.com/semgrep/semgrep-rules/blob/fb248f4c8ecd0474594180c1187fba068932cf84/python/sqlalchemy/security/sqlalchemy-execute-raw-query.yaml#L43-L48